### ISSUE TITLE
Request for full code

### ISSUE THREAD

**Issue by BassantTolba1234** [kclip/meta-autoencoder-without-channel-model#1]
Dear sir, 
Brilliant work! I was wondering if you'd kindly be willing to share your full code ? 
I read that you said you will share it soon (by the end of April) but it is not uploaded 
I'm very interested in Meta Learning and I'd really like to try out some experiments with the Meta Learning. Thanks in Advance!

**Comment by sangwoo-p** [comment]
Hi, thank you for the attention of our work! I am terribly sorry for not keeping my words on my posting plan. I was digesting my full code and adding some comments which made the job took longer than I thought... I will speed up my process and promise to post it within three days :) Again, sorry for making you wait.

**Comment by BassantTolba1234** [comment]
I’m waiting for that... Thank you very much I really appreciate that.
Good Luck!

**Comment by sangwoo-p** [comment]
Hi, I just uploaded the full code :)
Thank you and sorry again for making you wait...
Please feel free to contact me anytime if you have any queries regarding to the codes and/or the paper!

**Comment by BassantTolba1234** [comment]
Hi
Amazing work  and great efforts !
Thank you a million .
I hope you all the best and success insha Allah.

**Comment by sangwoo-p** [comment]
Hi, after my first upload, I found some minor bugs of my code that may occur some errors.
The essential part of the code is unchanged but please check the latest version that I have fixed all the bugs that I have found. (Basically, the bugs came from digesting my code and renaming the variables which made the code unstable in the first version.)
Sorry for causing such inconvenience due to my incompleteness.
I believe I can close this issue now and please let me know if you have any more issues :)
Hope you too all the best and success insha Allah!

**Comment by BassantTolba1234** [comment]
Hi , never mind sure. good efforts and good work.
PLease i have some questions.

1- Does your paper (End-to-End Fast Training of Communication Links Without
a Channel Model via Online Meta-Learning
) make Channel estimation and Feedback ?? or it improves the End To End
training schemes ??

2- Is your paper the same as this paper (META-LEARNING TO COMMUNICATE: FAST
END-TO-END TRAINING FOR FADING CHANNELS
here is its link (https://arxiv.org/abs/1910.09945 )?? ,
I mean the two papers are the same , or they are different?

3-I'm very interested in channel estimation and feedback using Meta
Learning. So kindly could you share with me the code of this paper
(RoemNet: Robust Meta Learning based Channel Estimation in OFDM Systems
,here is the link(https://ieeexplore.ieee.org/document/8761319)
)you referenced in your paper .


I'm waiting for your reply.
Really thank you in advance for your help.
I hope you all the best always and more insha Allah.

**Comment by sangwoo-p** [comment]
Hi, here comes my answer.

1) My paper assumes feedback without direct channel estimation. As we
consider two neural networks: one for transmitter and the other for
receiver, and the transmitter gets bits as input while output as
transmitted symbol and the receiver gets received signal as input while
output as estimated bits, there is no explicit channel estimation
procedure. The feedback is calculated at the receiver side by calculating
cross entropy loss. The description on the feedback came from the paper [F.
A. Aoudia and J. Hoydis, “Model-free training of end-to-end communication
systems,” arXiv preprint arXiv:1812.05929, 2018] and you can find detailed
explanation in this paper. In short, as we consider hybrid scheme that
assumes no update at the transmitter side while receiver adapts based on
the available pilots, I can say that our scheme improves end-to-end
performance by adapting receiver based on the current channel's pilots in
which channel estimation may be implicitly done at the receiver side via
updating the neural network.

2) The two papers are different. Most different part is that one assumes
available channel model while the other assumes unavailable channel model.
The same difference can be found in the following two papers: [T. O’Shea
and J. Hoydis, “An introduction to deep learning for the physical layer,”
IEEE Trans. Cognitive Commun. and Netw., vol. 3, no. 4, pp. 563–575, Dec.
2017.] and [F. A. Aoudia and J. Hoydis, “Model-free training of end-to-end
communication systems,” arXiv preprint arXiv:1812.05929, 2018].

3) I'm afraid that I do not have the code for that paper since I haven't
tried to implement their work...

Please let me know if you have any more queries!

Hope you too all the best insha Allah!

With best regards,
Sangwoo Park


2020년 5월 25일 (월) 오후 7:08, BassantTolba1234 <notifications@github.com>님이 작성:

**Comment by BassantTolba1234** [comment]
Dear sir,
Really thank you very much for your clarification and for your continuous help .
I hope you achieve high levels of success insha Allah.
Thanks again.
Best Regards always,
